### PR TITLE
Extend paragraph on Mocking

### PR DESCRIPTION
**Pull request by holsky**
Here's my first draft to address #593 . What do you think? Is it factually correct, too much, too little?

### CODE REVIEW

**Comment by scouten** [comment]
As the person who raised the original issue (#593), this looks good now that the correct commit is associated with this PR 😆 .

Thanks!

File: lessons/basics/testing.md [hunk 1]
```diff
@@ -102,6 +102,11 @@ end
 
 ## Mocking
 
-The simple answer to mocking in Elixir: don't.  You may instinctively reach for mocks but they are highly discouraged in the Elixir community and for good reason.  If you follow good design principles the resulting code will be easy to test as individual components.
+The simple answer to mocking in Elixir is: don't.  You may instinctively reach for mocks but they are highly discouraged in the Elixir community and for good reason. 
+
+For a longer discussion there is this [excellent article](http://blog.plataformatec.com.br/2015/10/mocks-and-explicit-contracts/). The gist is, that instead of mocking away dependencies for testing (mock as a *verb*), it has many advantages to explicitly define interfaces (behaviors) for code outside your application and using Mock (as a *noun*) implementations in your client code for testing.
+
+To switch the implementations in your application code, the preferred way is to pass the module as arguments and use a default value. If that does not work, use the built-in configuration mechanism. For creating these mock implementations, you even do not need a mocking library, only behaviours and callbacks.
```
Comment (holsky) [inline lessons/basics/testing.md]: Sounds much better

File: lessons/basics/testing.md [hunk 2]
```diff
@@ -102,6 +102,11 @@ end
...
+The simple answer to mocking in Elixir is: don't.  You may instinctively reach for mocks but they are highly discouraged in the Elixir community and for good reason. 
+
+For a longer discussion there is this [excellent article](http://blog.plataformatec.com.br/2015/10/mocks-and-explicit-contracts/). The gist is, that instead of mocking away dependencies for testing (mock as a *verb*), it has many advantages to explicitly define interfaces (behaviors) for code outside your application and using Mock (as a *noun*) implementations in your client code for testing.
+
+To switch the implementations in your application code, the preferred way is to pass the module as arguments and use a default value. If that does not work, use the built-in configuration mechanism. For creating these mock implementations, you even do not need a mocking library, only behaviours and callbacks. 
+
+There are valid use cases for mocking libraries like [Mock](https://github.com/jjh42/mock). For example they can help you more easily create modules and objects that your application works with (not external entities), or help you verify if calls were made in the correct order and frequency.
```
Comment (doomspork) [inline lessons/basics/testing.md]: Sounds great @holsky, thank you 👍

**Comment by holsky** [comment]
I amended the commit and did a force push -- does that work well with how GitHub handles code review?

**Pull request merged**